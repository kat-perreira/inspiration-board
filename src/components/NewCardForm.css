/* The Modal (background) */

.new-card-form {
  width: 50%;
  margin: auto;
  padding-bottom: 4rem;
  border: 5px solid black;
  justify-content: space-around;
}

.new-card-form__header {
  text-align: center;
  text-transform: uppercase;
}

.new-card-form__form {
  font-size: 1.5em;
  display: grid;
  grid-template-columns: [labels] auto [controls] 3fr;
  grid-auto-flow: row;
  grid-gap: .11em;
}

.new-card-form__form-label  {
  grid-column: labels;
  grid-row: auto;
}

.new-card-form__form-select,
.new-card-form__form-textarea,
.new-card-form__form-button {
  font-family: 'Raleway', sans-serif;
  font-size: 1.25em;
  grid-column: controls;
  grid-row: auto;
}

.new-card-form__form-button {
  background-color: inherit;
  border: 1px solid #222222;
  font-size: 1em;
}
