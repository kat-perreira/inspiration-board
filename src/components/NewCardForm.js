import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog", "sad_cat"]

class NewCardForm extends Component{
  constructor(props) {
    super(props);

    this.state = {
      text: "",
      emoji: "",

    };
  }

  resetState = () => {
    this.setState({
      text: "",
      emoji: "",
      showMenu: false,
    });
  }

  onFormChange = (event) => {
    const text = event.target.name;
    const value = event.target.value;

    const updatedState = {};
    updatedState[text] = value;
    this.setState(updatedState);
  }

  onSubmit = (event) => {
    event.preventDefault();
    const { text, emoji } = this.state;



    console.log(event);
    this.props.addCardCallback(this.state);
    this.resetState();
  }

// wrie a closure to return list of emojis
  emojiOptions = () => {
    const listEmojis = EMOJI_LIST.map((word, i) => {
      return (
        <option
          key={i}
          name="emoji"
          value={word}
          >
          {emoji.getUnicode(`${word}`)}
        </option>)
    });
    return[<option name="board" value="" key="emoji-select">Select Emoji</option>, listEmojis]
  };

  render() {

    return (

      <div className="new-card-form">
        <form onSubmit={this.onSubmit} className="new-card-form__form">
          <section className="new-card-form__header" >
            <h1>Add Some Inspiration</h1>
          </section>

          <label
            className="new-card-form__form-label"
            htmlFor="Text"
            >text</label>


          <input
              className="new-card-form_form-textarea"
              name="text"
              placeholder="Inspirational Message"
              onChange={this.onFormChange}
              value={this.state.text}>
            </input>



          <div>
            <label
              className="new-card-form__form-label"
              htmlFor="Emoji">Emoji
            </label>

            <select
              name="emoji"
              id="emoji"
              key={this.props.value}
              value={this.state.word}
              onChange={this.onFormChange}
              >
             { this.emojiOptions() }
            </select>


          <input
            className="new-card-form__form-button"
            type="submit"
            name="submit"
            value="Add a Card"
            onSubmit={this.handleSubmit}
            />


      </div>
    </form>
  </div>
    );
  }


}


NewCardForm.propTypes = {
  addCardCallback: PropTypes.func.isRequired,
};

export default NewCardForm;
