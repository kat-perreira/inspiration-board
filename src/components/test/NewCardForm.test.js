import React from 'react';
import NewCardForm from '../NewCardForm';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('that it renders when passed an empty function', () => {
    const wrapper = shallow(
      <NewCardForm
        addCardCallback={() => {}}/>);

    expect(wrapper).toMatchSnapshot();
  });

});
