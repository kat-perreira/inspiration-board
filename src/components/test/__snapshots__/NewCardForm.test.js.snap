// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it renders when passed an empty function 1`] = `
<div
  className="new-card-form"
>
  <form
    className="new-card-form__form"
    onSubmit={[Function]}
  >
    <section
      className="new-card-form__header"
    >
      <h1>
        Add Some Inspiration
      </h1>
    </section>
    <label
      className="new-card-form__form-label"
      htmlFor="Text"
    >
      text
    </label>
    <input
      className="new-card-form_form-textarea"
      name="text"
      onChange={[Function]}
      placeholder="Inspirational Message"
      value=""
    />
    <div>
      <label
        className="new-card-form__form-label"
        htmlFor="Emoji"
      >
        emoji
      </label>
      <select
        id="emoji"
        name="emoji"
        onChange={[Function]}
      >
        <option
          key="emoji-select"
          name="board"
          value=""
        >
          Select Emoji
        </option>
        <option
          key="0"
          name="emoji"
          value=""
        />
        <option
          key="1"
          name="emoji"
          value="heart_eyes"
        >
          😍
        </option>
        <option
          key="2"
          name="emoji"
          value="beer"
        >
          🍺
        </option>
        <option
          key="3"
          name="emoji"
          value="clap"
        >
          👏
        </option>
        <option
          key="4"
          name="emoji"
          value="sparkling_heart"
        >
          💖
        </option>
        <option
          key="5"
          name="emoji"
          value="heart_eyes_cat"
        >
          😻
        </option>
        <option
          key="6"
          name="emoji"
          value="dog"
        >
          🐶
        </option>
        <option
          key="7"
          name="emoji"
          value="sad_cat"
        />
      </select>
      <input
        className="new-card-form__form-button"
        name="submit"
        type="submit"
        value="Add a Card"
      />
    </div>
  </form>
</div>
`;
